packages/itwinui-react/src/core/Modal/Modal.tsx

```diff
@@ -99,10 +114,11 @@ export const Modal = React.forwardRef((props, forwardedRef) => {
       setFocus
       ref={forwardedRef}
```

Comment:
This part is very annoying: `ref` and `...rest` props are applied on different elements. I thought we fixed it in v3, but evidently not.